app/models/role.rb

```diff
@@ -0,0 +1,29 @@
+# == Schema Information
+#
+# Table name: roles
+#
+#  id            :bigint           not null, primary key
+#  name          :string
+#  resource_type :string
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  resource_id   :bigint
+#
+# Indexes
+#
+#  index_roles_on_name_and_resource_type_and_resource_id  (name,resource_type,resource_id)
+#  index_roles_on_resource                                (resource_type,resource_id)
+#
+class Role < ApplicationRecord
+  has_and_belongs_to_many :staff_accounts, join_table: :staff_accounts_roles
+
+  belongs_to :resource,
+    polymorphic: true,
+    optional: true
+
+  validates :resource_type,
+    inclusion: {in: Rolify.resource_types},
+    allow_nil: true
+
+  scopify
```

Comment:
What is this additional `scopify` for?

Reply:
the entire class was created by the gem after running the rolify generator - I did not add scopify

Reply:
Gotcha, i'll assume it has a purpose and we can keep it as such.